### PR TITLE
Backend songs

### PR DESCRIPTION
**Pull request by marcelarie**
**1. Build song controller methods for:**
  - Get all songs.
  - Get all songs by parameters.
  - Get song by name.
  - Post song.
  - Patch song.
  - Always return updated document on patch.
 
**2. Build playlist and song mongoose model.**
**3. Build Song Router.**

### CODE REVIEW

File: packages/api/src/controllers/song-controller/index.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+}
+
+export {
+    getAllSongs,
+    getSongByName,
+    getSongsByParams,
+    postSong,
+    patchSongByName,
+};
```
Comment (marcelarie) [inline packages/api/src/controllers/song-controller/index.js:96]: Song controller.

File: packages/api/src/models/playlist-model.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        {
+            type: moongoose.Schema.Types.ObjectId,
+            ref: 'song', // (?) Song
+        },
+    ],
+});
+
+const Playlist = moongoose.model('Playlist', PlaylistSchema);
+
+export default Playlist;
```
Comment (marcelarie) [inline packages/api/src/models/playlist-model.js:18]: Playlist Schema

File: packages/api/src/models/song-model.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        require: true,
+    },
+    // video: { (?)
+    //     type: String,
+    // }
+});
+
+const Song = mongoose.model('song', SongSchema);
+
+export default Song;
```
Comment (marcelarie) [inline packages/api/src/models/song-model.js:42]: Song Schema

File: packages/api/src/repositories/song-repository.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    find: body => {
+        return normalizeDBQuery(Song.find(body));
+    },
+
+    findOne: filter => {
+        return normalizeDBQuery(Song.findOne(filter, '-__v'));
+    },
+
+    //                      return the updated document ↴
+    findOneAndUpdate: (filter, body, option = { new: true }) => {
```
Comment (marcelarie) [inline packages/api/src/repositories/song-repository.js:18]: Always udpated document on return of patch.

File: packages/api/src/routes/song-router.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+const songRouter = Router();
+
+songRouter.use(authMiddleware); // (?)
+
+//                      (?) by id
+songRouter.get('/song/:name', getSongByName);
+songRouter.get('/songs/all', getAllSongs);
+songRouter.post('/songs/all-with', getSongsByParams);
+songRouter.post('/song', postSong);
+songRouter.patch('/song/:name', patchSongByName);
```
Comment (marcelarie) [inline packages/api/src/routes/song-router.js:22]: Song Router paths.

**Comment by miguelgc95** [comment]
Nice work!

**Pull request merged**